src/components/__tests__/NavBar.test.tsx

```diff
@@ -0,0 +1,83 @@
+import React from 'react';
+import { render, screen, fireEvent } from '@testing-library/react';
+import { NavBar } from 'components/NavBar';
+import mockRouter from 'next-router-mock';
+import { ThemeProvider, createTheme } from '@mui/material/styles';
+
+jest.mock('next/router', () => require('next-router-mock'));
+
+const theme = createTheme();
+
+describe('NavBar', () => {
+  beforeEach(() => {
+    mockRouter.setCurrentUrl('/');
+  });
+
+  const renderWithRouter = (ui: React.ReactElement) => {
+    return render(<ThemeProvider theme={theme}>{ui}</ThemeProvider>);
+  };
+
+  it('should render logo and menu items', () => {
+    renderWithRouter(<NavBar />);
+    
+    expect(screen.getByAltText('Logo')).toBeInTheDocument();
+    expect(screen.getByText('Find a mentor')).toBeInTheDocument();
+    expect(screen.getByText('Programmes')).toBeInTheDocument();
+    expect(screen.getByText('About Us')).toBeInTheDocument();
+    expect(screen.getByText('Jobs')).toBeInTheDocument();
+    expect(screen.getByText('Events')).toBeInTheDocument();
+    expect(screen.getByText('Blog')).toBeInTheDocument();
+  });
+
+  it('should navigate to the correct path on menu item click', () => {
+    renderWithRouter(<NavBar />);
+
+    fireEvent.click(screen.getByText('Programmes'));
+    expect(screen.getByRole('menuitem', { name: /Book Club/i })).toBeInTheDocument();
+    expect(screen.getByText('Book Club')).toBeInTheDocument();
+    fireEvent.click(screen.getByText('Book Club'));
+    expect(mockRouter).toMatchObject({ pathname: '/programmes/book-club' });
+
+    fireEvent.click(screen.getByText('Programmes'));
+    expect(screen.getByRole('menuitem', { name: /Our Programmes/i })).toBeInTheDocument();
+    expect(screen.getByText('Our Programmes')).toBeInTheDocument();
+    fireEvent.click(screen.getByText('Our Programmes'));
+    expect(mockRouter).toMatchObject({ pathname: '/programmes' });
+
+    fireEvent.click(screen.getByText('About Us'));
+    expect(screen.getByRole('menuitem', { name: /Team/i })).toBeInTheDocument();
```

Comment:
dummy question what is this "i" /Team/i here?

Reply:
its a flag that stands for case insensitive for example when writing regular expressions. it can basically match words like Team, TEAM,